cql3/restrictions/restriction.hh

```diff
@@ -108,6 +113,15 @@ struct conjunction {
 /// directly into the resulting conjunction's children, flattening the expression tree.
 extern ::shared_ptr<expression> make_conjunction(::shared_ptr<expression> a, ::shared_ptr<expression> b);
 
+/// Checks if restr is satisfied by the given data, then throws if the result is different from
+/// expected.
+extern void check_is_satisfied_by(
+        shared_ptr<expression> restr,
```

Comment:
`const expression&`? No need to bump reference counts, and we can indicate this function doesn't modify restr by using const.

Reply:
Done.